docs/tasks/manage-stateful-set/upgrade-pet-set-to-stateful-set.md

```diff
@@ -0,0 +1,152 @@
+kubectl get petsets --all-namespaces -o yaml > all-petsets.yaml
+```
+
+#### Prepare Stateful Set manifests 
+
+Now, for every Pet Set manifest you have, prepare a corresponding Stateful Set manifest with the following steps: 
+
+1. Change `apiVersion` from `apps/v1alpha1` to `apps/v1beta1`
+2. Change `kind` from `PetSet` to `StatefulSet`
+3. If you have the debug hook annotation `pod.alpha.kubernetes.io/initialized` set to `true`, you may remove it since it's redundant. If you don't have this annotation, you should add one, with the value set to `false`, to pause Stateful Sets operations.
+
+It's recommended that you keep both Pet Set manifests and Stateful Set manifests, so that you may safely rollback and recreate your Pet Sets, 
+if you decide not to upgrade your cluster. 
+
+#### Delete all Pet Sets without cascading
+
+If you find existing Pet Sets in your cluster in the previous step, you need to delete all Pet Sets *without cascading*. You can do this from `kubectl` with `--cascade=false`. 
+Note that if the flag wasn't set, **cascading deletion will be performed by default**, and all pods managed by your Pet Sets will be gone. 
+
+You may delete those Pet Sets by specifying file names. This only works when in
+the files there are only Pet Sets, but not other resources (such as Services):
+
+```shell
+# Delete all existing Pet Sets without cascading 
+# Note that <pet-set-file> should only contain Pet Sets that you want to delete, but not any other resources
+kubectl delete -f <pet-set-file> --cascade=false
+```
+
+Alternatively, you may delete them by specifying resource names: 
+
+```shell
+# Alternatively, delete them by name and namespace without cascading
+kubectl delete petsets <pet-set-name> -n=<pet-set-namespace> --cascade=false
+```
+
+To make sure you've deleted all Pet Sets in the system, run this command, which
+should return nothing:
+
+```shell
+# Get all Pet Sets again to make sure you deleted them all 
+kubectl get petsets --all-namespaces
+```
+
+At this moment, you've deleted all Pet Sets in your cluster, but not their Pods, Persistent Volumes, or Persistent Volume Claims. 
+However, since the pods are not managed by Pet Sets anymore, they will be vulnerable to node failures until you finish the master upgrade and recreate Stateful Sets.
+
+#### Upgrade Kubernetes master to >= 1.5
+
+Now, you may [upgrade your Kubernetes master](/docs/admin/cluster-management/#upgrading-a-cluster) to >= 1.5.0.
+Note that Kubernetes nodes should NOT be upgraded at this point, since the pods
+(that were once managed be Pet Sets) are now vulnerable to node failures. 
+
+#### Upgrade kubectl to >= 1.5
+
+Upgrade `kubectl` to >= 1.5 following [the steps for installing and setting up 
+kubectl](/docs/user-guide/prereqs/).
+
+#### Create Stateful Sets
+
+Make sure you have both master and `kubectl` upgrade to a version >= 1.5 before
+continuing. To verify this, run this command to make sure both client and server
```

Comment:
This is because the command output only says "Client Version" and "Server Version". I updated this part to make it more clear.